modules/library/main/src/main/java/org/geotools/styling/RasterSymbolizerImpl.java

```diff
@@ -425,10 +422,12 @@ public void setOpacity(Expression opacity) {
      */
     @Deprecated
     public void setOverlap(Expression overlap) {
-        if (this.overlap == overlap) {
+        if(overlap == null)	{
             return;
         }
-        this.overlap = overlap;
+
+        OverlapBehavior overlapBehavior = OverlapBehavior.valueOf(overlap.toString());
```

Comment:
This assumes the expression is a literal, the specification says this must evaluate to one of four literals.

To fix this use overlap.evaulate( null, String.class ) - it will work for literal, env function etc...